app/_hub/kong-inc/json-threat-protection/overview/_index.md

```diff
@@ -0,0 +1,258 @@
+---
+nav_title: Overview
+---
+
+The JSON Threat Protection plugin provides security validation against various aspects of JSON structure.
+The plugin validates the incoming JSON request body to ensure the payload adheres to the policy limits, regardless of whether the `Content-Type` header exists or is set to `application/json`.
+Requests violating the policy are considered malicious.
+You can configure the plugin to drop such requests, which stops them from reaching the service.
+Optionally, the plugin can operate in tap mode to monitor the traffic.
+
+## How it works
+
+The plugin checks the following limits:
+
+- Maximum container depth of the entire JSON
+- Maximum number of array elements
+- Maximum number of object entries
+- Maximum length of object keys
+- Maximum length of strings
+
+For example, for the following JSON:
+
+```
+{
+  "name": "Jason",
+  "age": 20,
+  "gender": "male",
+  "parents": ["Joseph", "Viva"]
+}
+```
+
+- Maximum container depth: 2
+- Maximum number of array elements: 2
+- Maximum number of object entries: 4
+- Maximum length of object keys: 7 (`parents`)
+- Maximum length of strings: 6 (`Joseph`)
+
+{:.note}
+> **Note**: Length calculation for JSON strings and object entry names is based on UTF-8 characters, not bytes.
+
+
+Additionally, there is a limit named `max_body_size`, which is used to restrict the request body size.
+This can effectively reduce resource overhead and potential attack risks associated with excessively large bodies.
+The plugin reads the `Content-Length` header and compares its value with `max_body_size`.
+Therefore, in block mode, if the `Content-Length` header is missing or its value exceeds `max_body_size`, the request will be terminated.
+In tap mode, only the body size is checked, and logs are recorded.
+
+{:.note}
+> **Note**: max_body_size and nginx_http_client_max_body_size are independent of each other. Therefore, users should be aware that if nginx_http_client_max_body_size is set to a larger value while max_body_size is smaller and block mode is enabled, any request with a body size greater than max_body_size but less than nginx_http_client_max_body_size will be terminated.
```

Comment:
```suggestion
> **Note**: `max_body_size` and `nginx_http_client_max_body_size` are independent of each other. Therefore, if `nginx_http_client_max_body_size` is set to a larger value while `max_body_size` is smaller and block mode is enabled, any request with a body size greater than `max_body_size` but less than `nginx_http_client_max_body_size` will be terminated.
```
Sticking the params in backticks to appease the linter, and removing some extra wordiness.